dot/peerset/peerset.go

```diff
@@ -332,21 +349,24 @@ func (ps *PeerSet) reportPeer(change ReputationChange, peers ...peer.ID) error {
 
 		setLen := ps.peerState.getSetLength()
 		for i := 0; i < setLen; i++ {
-			if ps.peerState.peerStatus(i, pid) == connectedPeer {
-				// disconnect peer
-				err = ps.peerState.disconnect(i, pid)
-				if err != nil {
-					return err
-				}
+			if ps.peerState.peerStatus(i, pid) != connectedPeer {
+				continue
+			}
 
-				ps.resultMsgCh <- Message{
-					Status: Drop,
-					setID:  uint64(i),
-					PeerID: pid,
-				}
-				if err = ps.allocSlots(i); err != nil {
-					return err
-				}
+			// disconnect peer
+			err = ps.peerState.disconnect(i, pid)
+			if err != nil {
+				return fmt.Errorf("cannot disconnect peer %s at set %d: %w", pid, i, err)
+			}
+
+			ps.resultMsgCh <- Message{
+				Status: Drop,
+				setID:  uint64(i),
+				PeerID: pid,
+			}
+
+			if err = ps.allocSlots(i); err != nil {
+				return fmt.Errorf("could not allocate slots: %w", err)
```

Comment:
nit add peer id context perhaps?

```suggestion
				return fmt.Errorf("could not allocate slots for peer %s: %w", pid, err)
```

Reply:
Is not needed as the `ps.allocSlots`error already has the context of some peer